CRM/Donrec/Exporters/EmailPDF.php

```diff
@@ -140,6 +140,21 @@ protected function sendEmail($receipt, $pdf_file, $is_test, $snapshot_line_id) {
       // load contact data
       $contact = civicrm_api3('Contact', 'getsingle', array('id' => $receipt['contact_id']));
 
+      if (CRM_Donrec_Logic_Settings::get('email_location_type_id')) {
+        // load email address from the configured location type
+        $email_location_type_id = CRM_Donrec_Logic_Settings::get('email_location_type_id');
+        $email = \Civi\Api4\Email::get(FALSE)
+          ->addSelect('email')
+          ->addWhere('location_type_id', '=', $email_location_type_id)
+          ->addWhere('contact_id', '=', $contact['id'])
+          ->setLimit(1)
+          ->execute()
+          ->first();
+        if ($email) {
+          $contact['email'] = $email['email'];
+        }
```

Comment:
Since `$contact['email']` is being checked below anyway, the `if()` isn't necessary.

Reply:
This validation is to ensure that if no email is found, the process continues with the email selected by civicrm_api3.